File: tests/test_models/models/passthrough.py
##
# SPDX-License-Identifier: LGPL-2.1-only
#
# Copyright (C) 2019 Samsung Electronics
#
# @file    passthrough.py
# @brief   Python custom filter example: passthrough
# @author  Dongju Chae <dongju.chae@samsung.com>

import numpy as np
import nnstreamer_python as nns

D1 = 3
D2 = 280
D3 = 40
D4 = 1


##
# @brief  User-defined custom filter; DO NOT CHANGE CLASS NAME
class CustomFilter(object):
    ##
    # @brief  The constructor for custom filter: passthrough
    def __init__(self, *args):
        self.input_dims = [nns.TensorShape([D1, D2, D3, D4], np.uint8)]
        self.output_dims = [nns.TensorShape([D1, D2, D3, D4], np.uint8)]

    ##
    # @brief  python callback: getInputDim
    # @param  None
    # @return user-assigned input dimensions
    def getInputDim(self):
        return self.input_dims

    ##
    # @brief  Python callback: getOutputDim
    # @param  None
    # @return user-assigned output dimensions
    def getOutputDim(self):
        return self.output_dims

    ##
    # @brief  Python callback: invoke
    # @param  Input tensors: list of input numpy array
    # @return output tensors: list of output numpy array
    def invoke(self, input_array):
        # passthrough, just return.
        return input_array

File: tests/test_models/models/passthrough.lua
-- passthrough.lua

inputTensorsInfo = {
  num = 1,
  dim = {{3, 640, 480, 1}, },
  type = {"uint8", }
}

outputTensorsInfo = {
  num = 1,
  dim = {{3, 640, 480, 1}, },
  type = {"uint8", }
}

iC = inputTensorsInfo["dim"][1][1]
iW = inputTensorsInfo["dim"][1][2]
iH = inputTensorsInfo["dim"][1][3]
iN = inputTensorsInfo["dim"][1][4]

oC = outputTensorsInfo["dim"][1][1]
oW = outputTensorsInfo["dim"][1][2]
oH = outputTensorsInfo["dim"][1][3]
oN = outputTensorsInfo["dim"][1][4]

function nnstreamer_invoke()
  input = input_tensor(1)
  output = output_tensor(1)

  for i=1,oC*oW*oH*oN do
    output[i] = input[i]
  end
end
